modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -741,16 +741,18 @@
 (defn- str-match-pattern [field options prefix value suffix]
   (if (mbql.u/is-clause? ::not value)
     {$not (str-match-pattern field options prefix (second value) suffix)}
-    (let [base-type (get-in field [2 :base-type])]
+    (let [base-type (get-in field [2 :base-type])
+          supports-toString? (-> (get-mongo-version) :semantic-version (driver.u/semantic-version-gte [8]))]
```

Comment:
I think this should be a database feature like  `convert-uuid-to-string` and the version check happens in there. Ideally, metabase.lib would be able to drop uuids from `contains` etc if the feature doesn't exist but if that's too difficult for now we can throw the exception.